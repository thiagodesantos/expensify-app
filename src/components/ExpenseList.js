import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile">Despesas</div>
            <div className="show-for-desktop">Despesa</div>
            <div className="show-for-desktop">Valor</div>
        </div>  
        <div className="list-body">
        {
            props.expenses.length === 0 ? (
              <div className="list-item list-item--message">
                <span>Não há despesas</span>
              </div>
            ) : (
                props.expenses.map((expense) => {
                    return <ExpenseListItem key={expense.id} {...expense} />
                })
            )
        }    
        </div>        
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);