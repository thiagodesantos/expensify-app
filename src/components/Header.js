import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Despesinhas</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
        <NavLink to="/criar" activeClassName="is-active">Adicionar</NavLink>
        <NavLink to="/ajuda" activeClassName="is-active">Ajuda</NavLink>
    </header>
);

export default Header;