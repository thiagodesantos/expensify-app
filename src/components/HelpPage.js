import React from 'react';

const HelpPage = () => (
    <div> Esse Ã© o help component</div>
);

export default HelpPage;