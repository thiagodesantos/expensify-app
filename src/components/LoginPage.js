import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Despesinhas</h1>
            <p>Ã‰ hora de colocar suas despesas sobre controle!</p>
            <button className="button" onClick={startLogin}>Login com o Google</button>
        </div>        
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);