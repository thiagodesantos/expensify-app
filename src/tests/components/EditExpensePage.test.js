import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import expenses from '../fixtures/expenses';

let startEditExpense, startRemoveExpense, history, wrapper;

beforeEach(() =>{
    startEditExpense = jest.fn();
    startRemoveExpense = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(
        <EditExpensePage 
        startEditExpense={startEditExpense} 
            startRemoveExpense={startRemoveExpense} 
            history={history} 
            expense={expenses[2]}/>);
});

test('Deveria renderizar startEditExpense Corretamente', () => {   
    expect(wrapper).toMatchSnapshot();
});

test('Deveria lidar com o EditExpense', () => { 
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[2]);
    expect(history.push).toHaveBeenCalledWith('/');
    expect(startEditExpense).toHaveBeenCalledWith(expenses[2].id, expenses[2]);
});

test('Deveria lidar com o startRemoveExpense', () => { 
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenCalledWith('/');
    expect(startRemoveExpense).toHaveBeenCalledWith({
        id: expenses[2].id
    });
});