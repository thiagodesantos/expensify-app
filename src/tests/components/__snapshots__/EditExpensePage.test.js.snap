// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deveria renderizar startEditExpense Corretamente 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 4500,
        "createdAt": 345600000,
        "description": "CartÃ£o de credito",
        "id": "3",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remover
  </button>
</div>
`;
